### PR TITLE
correct spelling of type 'timestamp with time zone' in JDBC connector

### PR DESCRIPTION
**Pull request by xandris**
[`BaseJdbcClient.java`](../tree/master/presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java#L95) has a typo in the name of the SQL type 'timestamp with time zone'. To reproduce, try to make a table containing a timestamp in a PostgreSQL database via JDBC:

```sql
CREATE TABLE postgresql.public.foo AS SELECT now() as rightnow
```

PostgreSQL will complain about an unexpected `with` token since Presto translates this query into something like this:

```sql
CREATE TABLE "temp_abcdef" ("rightnow" timestamp with timezone)
```

It should instead be:

```sql
CREATE TABLE "temp_abcdef" ("rightnow" timestamp with time zone)
```

The type map appears to be in the `BaseJdbcClient` class. There is another reference to this spelling in the MySQL code, but this PR doesn't modify that code.

### CODE REVIEW

**Comment by electrum** [comment]
Thanks for fixing this! Can you add a test to `TestPostgreSqlIntegrationSmokeTest`?

**Comment by xandris** [comment]
@electrum would it be sufficient to add it to an existing unit test, like `testInsert`?

**Comment by xandris** [comment]
I'm running into a different issue now... The expression I'm using in my `CREATE TABLE AS` is statically typed to be a `TIMESTAMP WITH TIME ZONE` (`now()` or `current_timestamp`), so the table gets created with a column of that type, but ultimately gets sent to the parameterized query as a `BIGINT` during the `INSERT` phase of `CREATE TABLE AS`.

**Comment by xandris** [comment]
Oh, `JdbcRecordSink` doesn't support time/timetz/timestamp/timestamptz

**Comment by xandris** [comment]
So PostgreSQL doesn't store the timezone information even for the `TIMESTAMP WITH TIME ZONE` type. What is the philosophy with respect to date/time types? Should Presto just store them as a `BIGINT` with the 12-bit timezone code? That makes them fairly unusable to anyone else accessing the PostgreSQL database directly. Or should Presto store them as a UNIX timestamp or `TIMESTAMP [WITH TIME ZONE]` and discard the timezone information?

I guess the main question is: how should `CREATE TABLE` in Presto create date/time columns in PostgreSQL? And the answer to that informs how dates/times are stored into/retrieved from PostgreSQL.

**Comment by xandris** [comment]
Well if someone can take a look at this I'd really appreciate it. This turned out to be a bigger change than I anticipated. I'm sure there are some controversial changes or stupid mistakes

File: presto-base-jdbc/pom.xml [hunk 1]
```diff
@@ -18,6 +18,11 @@
 
     <dependencies>
         <dependency>
+            <groupId>com.facebook.presto</groupId>
+            <artifactId>presto-main</artifactId>
```
Comment (martint) [inline presto-base-jdbc/pom.xml]: Connectors shouldn't depend on presto-main for anything other than tests

Comment (xandris) [inline presto-base-jdbc/pom.xml]: `JdbcRecordSink` receives a packed date with a 12-bit time zone key and it needs to look up the `DateTimeZone` for that key. What's the proper way to do that without consulting `DateTimeZoneIndex`?

Comment (xandris) [inline presto-base-jdbc/pom.xml]: I think I got it

**Comment by xandris** [comment]
What do you think now that the POM was reverted?

**Comment by xandris** [comment]
What can I do to help get this merged?

File: presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java [hunk 2]
```diff
@@ -498,9 +498,9 @@ protected Type toPrestoType(int jdbcType, int columnSize)
             case Types.DATE:
                 return DATE;
             case Types.TIME:
-                return TIME;
+                return "timetz".equalsIgnoreCase(typeName) ? TIME_WITH_TIME_ZONE : TIME;
```
Comment (martint) [inline presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java]: Are "timetz" and "timestamptz" applicable to any database (i.e., defined by the JDBC spec?) or are they specific to postgresql? If the latter, this should go in `PostgreSqlClient`.

Comment (xandris) [inline presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java]: probably only postgresql...

**Comment by combineads** [comment]
@electrum Can you review?
I also need to insert timestamp type in netezza JDBC. It does not support long type in timestamp type.
Thank you for efforts.

File: presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java (hunk 2 above)
Comment (combineads) [reply presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java]: In netezza, timetz that is alias time with time zone is OK.
https://www.ibm.com/support/knowledgecenter/en/SSULQD_7.2.1/com.ibm.nz.sproc.doc/c_sproc_data_types_aliases.html

**Pull request closed without merging**